pandora-client-web/src/browserStorage.ts

```diff
@@ -38,23 +38,14 @@ export class BrowserStorage<T> extends Observable<T> {
 	public setParse(value: string | null): void {
 		if (value !== null) {
 			const parsedValue = JSON.parse(value) as unknown;
-			if (typeof this.validate === 'function') {
-				if (this.validate(parsedValue)) {
-					const c = this._saveInhibit;
-					this._saveInhibit = true;
-					this.value = parsedValue as T;
-					this._saveInhibit = c;
-				}
-			} else {
-				const validated = this.validate.safeParse(parsedValue);
-				if (validated.success) {
-					this.value = validated.data;
-				}
+			const validated = this.validate.safeParse(parsedValue);
+			if (validated.success) {
+				this.value = validated.data;
 			}
 		}
 	}
 
-	public static create<T>(name: string, defaultValue: T, validate?: ZodType<T, ZodTypeDef, unknown> | ((value: unknown) => boolean)): BrowserStorage<T> {
+	public static create<T>(name: string, defaultValue: T, validate: ZodType<T, ZodTypeDef, unknown>): BrowserStorage<T> {
```

Comment:
In this function the `validate ?? (() => true)` can be simplified to `validate`.

Reply:
I wish TS would scream from an invalid type on both side of `??` regardless if the first part is always non null